src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -532,187 +897,241 @@ contract StrategyManagerUnitTests is Test, Utils {
+        }
+        require(
+            stakerStrategyListLengthBefore - numPoppedStrategies == strategyManager.stakerStrategyListLength(staker),
+            "stakerStrategyListLengthBefore - numPoppedStrategies != strategyManager.stakerStrategyListLength(staker)"
+        );
     }
+}
 
-    function test_depositIntoStrategyRevertsWhenStrategyDepositFunctionReverts() external {
-        // replace 'dummyStrat' with one that always reverts
-        dummyStrat = StrategyBase(address(new Reverter()));
-
-        // whitelist the strategy for deposit
-        cheats.startPrank(strategyManager.owner());
-        IStrategy[] memory _strategy = new IStrategy[](1);
-        _strategy[0] = dummyStrat;
-        cheats.expectEmit(true, true, true, true, address(strategyManager));
-        emit StrategyAddedToDepositWhitelist(dummyStrat);
-        strategyManager.addStrategiesToDepositWhitelist(_strategy);
-        cheats.stopPrank();
+contract StrategyManagerUnitTests_addShares is StrategyManagerUnitTests {
+    function test_Revert_DelegationManagerModifier() external {
+        DelegationManagerMock invalidDelegationManager = new DelegationManagerMock();
+        cheats.expectRevert(bytes("StrategyManager.onlyDelegationManager: not the DelegationManager"));
+        invalidDelegationManager.addShares(strategyManager, address(this), dummyStrat, 1);
+    }
 
-        address staker = address(this);
-        IERC20 token = dummyToken;
-        uint256 amount = 1e18;
-        IStrategy strategy = dummyStrat;
+    function testFuzz_Revert_StakerZeroAddress(uint256 amount) external {
+        cheats.expectRevert(bytes("StrategyManager._addShares: staker cannot be zero address"));
+        delegationManagerMock.addShares(strategyManager, address(0), dummyStrat, amount);
+    }
 
-        cheats.startPrank(staker);
-        cheats.expectRevert();
-        strategyManager.depositIntoStrategy(strategy, token, amount);
-        cheats.stopPrank();
+    function testFuzz_Revert_ZeroShares(address staker) external {
+        cheats.assume(staker != address(0));
+        cheats.expectRevert(bytes("StrategyManager._addShares: shares should not be zero!"));
+        delegationManagerMock.addShares(strategyManager, staker, dummyStrat, 0);
     }
 
-    function test_depositIntoStrategyRevertsWhenStrategyDoesNotExist() external {
-        // replace 'dummyStrat' with one that does not exist
-        dummyStrat = StrategyBase(address(5678));
+    function testFuzz_AppendsStakerStrategyList(address staker, uint256 amount) external {
+        cheats.assume(staker != address(0) && amount != 0);
+        uint256 stakerStrategyListLengthBefore = strategyManager.stakerStrategyListLength(staker);
+        uint256 sharesBefore = strategyManager.stakerStrategyShares(staker, dummyStrat);
+        require(sharesBefore == 0, "Staker has already deposited into this strategy");
+        require(!_isDepositedStrategy(staker, dummyStrat), "strategy shouldn't be deposited");
```

Comment:
Nit: possible to not use contractions so GitHub highlighting syntax doesn't break?